server/controllers/lecture-controller.js

```diff
@@ -383,3 +452,53 @@ const sendEmailsForCancelledLecture = async (lectureId) => {
     console.log(`There was an error cancelling the lecture: ${err}`);
   }
 };
+
+sendCandidateToReserveChangeEmail = async (lectureId, studentId) => {
+  const result = await knex
+    .select(
+      { id: "user.id" },
+      { name: "user.name" },
+      { surname: "user.surname" },
+      { email: "user.email" },
+      { lectureName: "lecture.name" },
+      { lectureStart: "lecture.start" },
+      { lectureCourseName: "course.name" }
+    )
+    .from("lecture_booking")
+    .join("user", "lecture_booking.student_id", "=", "user.id")
+    .join("lecture", "lecture.id", "=", "lecture_booking.lecture_id")
+    .join("course", "lecture.course", "=", "course.id")
+    .where("lecture_booking.lecture_id", lectureId)
+    .andWhere("user.id", studentId)
+    .limit(1);
+
+  const emailSubject = "You are moved from candidate list to Reserved List";
+  const emailBody = (
+    name,
+    surname,
+    lectureName,
+    lectureCourseName,
+    lectureStart
+  ) => `Dear ${name} ${surname},<br/> \
+  Your request state for lecture ${lectureName} of the course ${lectureCourseName} scheduled for ${lectureStart} has changed from candidate to reserve. Now you can participate in this lecture,\
+  <br/>The PULSeBS Team`;
+
+  const queue = result.map((item) =>
+    emailController.sendMail(
+      item.email,
+      emailSubject,
+      emailBody(
+        item.name,
+        item.surname,
+        item.lectureName,
+        item.lectureCourseName,
+        item.lectureStart
+      )
+    )
+  );
+  try {
+    await Promise.all(queue);
```

Comment:
`queue` is an array containing just one Promise, so i think that just an `await` would be enough
```suggestion
    await queue[0];
```